package/MDAnalysis/analysis/hbonds/hbond_analysis.py

```diff
@@ -582,6 +583,7 @@ def __init__(self, universe, selection1='protein', selection2='all', selection1_
         self.distance_type = distance_type  # note: everything except 'heavy' will give the default behavior
         self.angle = angle
         self.traj_slice = slice(start, stop, step)
+        self.pbc = pbc
```

Comment:
Validate the dimensions when initializing `self.pbc`
```
self.pbc = pbc and flags['use_periodic_selections'] and all(self.u.dimensions[:3])
```

Reply:
The check seems false. If the `flags` variable is set false this would always evaluate to false and the pbc condition is never applied.

Reply:
To be honest, I am not very familiar with setting the `flags` but I thought the default for `flags['use_periodic_selections']` was `True`, according to [selection.py](https://github.com/MDAnalysis/mdanalysis/blob/d5dec2b6f2be00fac35ee8f9be0106a244d6cc45/package/MDAnalysis/core/__init__.py#L270). Am I misreading this?

Reply:
The default can be `True` but the flags are changeable at runtime of the python session. You cannot rely on the fact that it is True. At the same time a setting of `pbc` should always overwrite the flag. The usual way of doing this is 

```python
def function(..., pbc=None):
    if pbc is None:
        pbc = flags['...']
```

This way we only use the flags if no value is provided for the function.

Reply:
flags are being removed soon, so I wouldn't add them anywhere any more

Reply:
so should I only validate the dimensions and not the flags ?

Reply:
@navyakhare yeah ignore flags but if pbc=True do a sanity check like `all(u.dimensions[:3])` else some error about no dimension information